### PR TITLE
Add flag for force sync local package when installing

### PR DESCRIPTION
**Pull request by carolineechen**

### CODE REVIEW

**Comment by carolineechen** [comment]
* **#1619** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1619?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1619?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: runhouse/resources/hardware/cluster.py [hunk 1]
```diff
@@ -753,17 +754,28 @@ def install_packages(
                 package. (Default: ``None``)
             conda_env_name (str, optional): Name of conda env to install the package in, if relevant. If left empty,
                 defaults to base environment. (Default: ``None``)
+            force_sync_local (bool, optional): Whether to install the locally installed version of a package if found,
+                if the package already exists on the cluster. By default, the local version will be installed only if
+                the package does not already exist on the cluster. To force installing the local version in this case,
+                set to ``True``. (Default: ``False``)
```
Comment (BelSasha) [inline runhouse/resources/hardware/cluster.py:760]: wonder if there is a way to make the documentation of the flag a bit more clear? 
How about:
Whether to install the locally installed version of a package ~~if found~~, if the package already exists on the cluster. By default, the local version will be installed only if the package does not already exist on the cluster. To force installing the local version in ~~this~~ in any (?) case, set to ``True``. (Default: ``False``)

File: runhouse/resources/packages/package.py [hunk 2]
```diff
@@ -249,15 +250,17 @@ def _install(
                 provided without a ``node``, package will be installed on the head node. (Default: ``None``)
             conda_env_name (Optional[str]): Name of the conda environment to install the package on, if using SSH and
                 installing in a specific conda env that is not activated by default.
+            force_sync_local (bool, optional): Whether to install the locally installed version of a package if found,
+                if the package already exists on the cluster. (Default: ``False``)
```
Comment (BelSasha) [inline runhouse/resources/packages/package.py:254]: similar to the note above

**Review by BelSasha — APPROVED** [review]
left couple nit comments :)

**Pull request merged**